tests/lib/rules/forbid-prop-types.js

```diff
@@ -892,6 +1024,231 @@ ruleTester.run('forbid-prop-types', rule, {
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.instanceOf(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.oneOf(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.oneOf([]),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.oneOfType(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.oneOfType([]),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.objectOf(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.arrayOf(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.exact(),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
+    }],
+    errors: 1
+  }, {
+    code: [
+      'var Hello = createReactClass({',
+      '  propTypes: {',
+      '    retailer: PropTypes.exact({}),',
+      '  },',
+      '  render: function() {',
+      '    return <div />;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: [{
+      forbid: ['empty']
```

Comment:
this doesn't really make sense to me as a schema; i think it'd be better to have `forbidEmpty: true` alongside `forbid`.